File: typescript/6kyu/Counting_Duplicates.ts
export const duplicateCount = (text: string): number => {
  const textArray = text.toLowerCase().split("");
  const textObj: { [key: string]: number } = {};
  let result = 0;
  for (let el of textArray) {
    el in textObj ? (textObj[el] += 1) : (textObj[el] = 0);
  }
  for (let key in textObj) {
    if (textObj[key] > 0) result++;
  }
  return result;
};

File: rust/src/6kyu/Counting_Duplicates.rs
fn count_duplicates(text: &str) -> u32 {
    if text.len() < 1 {
        return 0;
    }
    let mut dupes = 0;
    let mut text = text.to_lowercase();
    let i = text.remove(0);
    if text.contains(i) {
        dupes += 1;
    }
    text = text.split(i).collect::<Vec<&str>>().join("");
    dupes += count_duplicates(&*text);
    dupes
}